src/AudioLibs/AudioSourceSD.h

```diff
@@ -36,21 +36,31 @@ class AudioSourceSD : public AudioSource {
   /// Default constructor
   AudioSourceSD(const char *startFilePath = "/", const char *ext = ".mp3", int chipSelect = PIN_CS, bool setupIndex=true) {
     start_path = startFilePath;
-    exension = ext;
+    extension = ext;
     setup_index = setupIndex;
+    spi = &SPI;
     cs = chipSelect;
   }
 
+  // Pass your own spi instance, in case you need a dedicated one
+  AudioSourceSD(const char *startFilePath = "/", const char *ext = ".mp3", SPIClass &spiInstance = SPI, bool setupIndex=true) {
```

Comment:
I'm not sure if this will work when calling `AudioSourceSD source()`. I'd rather create the 2nd constructor as 

```cpp
AudioSourceSD(SPIClass &spiInstance, const char *startFilePath = "/", const char *ext = ".mp3", bool setupIndex=true)`
```

so that there's no possible confusion when using.